numpy/distutils/tests/test_system_info.py

```diff
@@ -0,0 +1,190 @@
+        return info
+
+class test_temp1(test_system_info):
+    section = 'temp1'
+class test_temp2(test_system_info):
+    section = 'temp2'
+
+class TestSystemInfoReading(TestCase):
+
+    def setUp(self):
+        """ Create the libraries """
+        # Create 2 sources and 2 libraries
+        self._dir1 = mkdtemp()
+        self._src1 = os.path.join(self._dir1,'foo.c')
+        self._lib1 = os.path.join(self._dir1,'libfoo.so')
+        self._dir2 = mkdtemp()
+        self._src2 = os.path.join(self._dir2,'bar.c')
+        self._lib2 = os.path.join(self._dir2,'libbar.so')
+        # Update local site.cfg
+        global simple_site, site_cfg
+        site_cfg = simple_site.format(**{
+                'dir1' : self._dir1 ,
+                'lib1' : self._lib1 ,
+                'dir2' : self._dir2 ,
+                'lib2' : self._lib2 
+                })
+        # Write the sources
+        with open(self._src1,'w') as fd:
+            fd.write(fakelib_c_text)
+        with open(self._src2,'w') as fd:
+            fd.write(fakelib_c_text)
+
+    def tearDown(self):
+        try: 
+            shutil.rmtree(self._dir1)
+            shutil.rmtree(self._dir2)
+        except: 
+            pass
+
+    def test_all(self):
+        """ Read in all information in the ALL block """
+        tsi = get_class('default')
+        assert_equal(tsi.get_lib_dirs(),[self._dir1,self._dir2])
+        assert_equal(tsi.get_libraries(),[self._lib1,self._lib2])
+        assert_equal(tsi.get_runtime_lib_dirs(),[self._dir1])
+        extra = tsi.calc_extra_info()
+        assert_equal(extra['extra_compile_args'],['-I/fake/directory'])
+
+    def test_temp1(self):
+        """ Read in all information in the temp1 block """
+        tsi = get_class('temp1')
+        assert_equal(tsi.get_lib_dirs(),[self._dir1])
+        assert_equal(tsi.get_libraries(),[self._lib1])
+        assert_equal(tsi.get_runtime_lib_dirs(),[self._dir1])
+
+    def test_temp2(self):
+        """ Read in all information in the temp2 block """
+        tsi = get_class('temp2')
+        assert_equal(tsi.get_lib_dirs(),[self._dir2])
+        assert_equal(tsi.get_libraries(),[self._lib2])
+        extra = tsi.calc_extra_info()
+        assert_equal(extra['extra_link_args'],['-Wl,-rpath='+self._lib2])
+
+    def test_compile1(self):
+        """ Compile source and link the first source """
+        tsi = get_class('temp1')
+        c = ccompiler.new_compiler()
+        # Change directory to not screw up directories
+        try:
+            previousDir = os.getcwd()
+        except OSError:
+            return
+        os.chdir(self._dir1)
+        c.compile([os.path.basename(self._src1)], output_dir=self._dir1)
+        # Ensure that the object exists
+        assert_(os.path.isfile(self._src1.replace('.c','.o')))
+        os.chdir(previousDir)
+
+    def test_compile2(self):
+        """ Compile source and link the second source """
```

Comment:
Can you change all test docstrings to comments? Docstrings mess up the verbose output of ``nose``, i.e. it shows the docstring instead of ``test_file.test_name``. The latter is much easier to locate when searching where a test failure comes from.